### PR TITLE
reworked API

### PR DESCRIPTION
**Pull request by kcajmagic**

### CODE REVIEW

File: webhookclient/argus.go [hunk 1]
```diff
@@ -116,30 +120,20 @@ func determineTokenAcquirer(config ArgusConfig) (acquire.Acquirer, error) {
...
-	for key, value := range filter {
-		buf.WriteString(key + "," + value)
-		buf.WriteString(",")
-	}
-	buf.Truncate(buf.Len() - 1)
-	return buf.String()
-}
 
-func (c *ArgusClient) GetWebhook() ([]webhook.W, error) {
+func (c *ArgusClient) GetWebhook(owner string) ([]webhook.W, error) {
```
Comment (kristinaspring) [inline webhookclient/argus.go:124]: I think you should remove any mention of argus in file/variable/function names from this webhookclient package.

File: webhookclient/argus.go [hunk 2]
```diff
@@ -178,20 +172,45 @@ func (c *ArgusClient) GetWebhook() ([]webhook.W, error) {
...
 	}
-	request, err := http.NewRequest("POST", fmt.Sprintf("%s/store/%s/%s%s", c.config.Address, c.config.Bucket, id, createAttributeFilter(c.config.Filters)), bytes.NewReader(data))
+	webhookPayload := map[string]interface{}{}
+	err = json.Unmarshal(webhookData, &webhookPayload)
+	if err != nil {
+		return err
+	}
+
+	var ttl int64
+	if int64(w.Duration.Seconds()) == 0 {
```
Comment (kristinaspring) [inline webhookclient/argus.go:187]:
```suggestion
	if int64(w.Duration.Seconds()) < 1 {
```

File: store/endpoint.go [hunk 3]
```diff
@@ -57,47 +60,68 @@ func (bre BadRequestError) StatusCode() int {
...
 		)
-		if kv, ok = request.(KeyItemPair); !ok {
+		if kv, ok = request.(KeyItemPairRequest); !ok {
 			return nil, BadRequestError{Request: request}
 		}
-		err := s.Push(kv.Key, kv.Item)
+		if kv.Identifier == "" {
+			return nil, BadRequestError{Request: request}
+		}
+		if kv.TTL == 0 {
```
Comment (kristinaspring) [inline store/endpoint.go:72]:
```suggestion
		if kv.TTL < 1 {
```

File: store/cassandra/executer_test.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
+package cassandra
+
+//
+// import (
```
[the remark this answers is not in the document]
Comment (kcajmagic) [reply store/cassandra/executer_test.go:4]: this file depends on running yugabyte cluster locally. I was going to delete the file, but decided against it.

**Pull request merged**